src/main/java/net/blay09/mods/cookingforblockheads/container/comparator/ComparatorSoL.java

```diff
@@ -40,9 +45,31 @@ public int compare(ItemStack o1, ItemStack o2) {
         } else if (!isFoodSecond) {
             return -1;
         }
-
-        boolean everEatenFirstFood = foodHistory.hasEverEaten(o1);
-        boolean everEatenSecondFood = foodHistory.hasEverEaten(o2);
+        boolean everEatenFirstFood = false, everEatenSecondFood = false;
+        if (!spiceCompat) {
+            everEatenFirstFood = foodHistory.hasEverEaten(o1);
+            everEatenSecondFood = foodHistory.hasEverEaten(o2);
+        } else {
+            // Older versions of Spice of Life don't have full food history, and therefore don't have a hasEverEaten
+            // This has similar, though not identical, behavior for those older versions.
+            try {
+                Method method = foodHistory.getClass().getDeclaredMethod("getHistory");
+                Object history = method.invoke(foodHistory);
+                FoodQueue foodQueue = (FoodQueue) history;
+                for (FoodEaten foodEaten : foodQueue) {
+                    if (foodEaten.itemStack.isItemEqual(o1)) {
+                        everEatenFirstFood = true;
+                    }
+                    if (foodEaten.itemStack.isItemEqual(o2)) {
+                        everEatenSecondFood = true;
+                    }
+                }
```

Comment:
Please use a [MethodHandle](https://docs.oracle.com/javase/8/docs/api/java/lang/invoke/MethodHandle.html) and cache it.